catalog/plugin/catalog-plugin-metacardingest-network/src/test/java/org/codice/ddf/catalog/plugin/metacard/util/KeyValueParserTest.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.metacard.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests the behavior of {@link KeyValueParser}.
+ */
+@RunWith(JUnit4.class)
+public class KeyValueParserTest {
+
+    private static final List<String> TEST_CASES = Arrays.asList("name=factoryName",
+            "ser=vice=3.2.1",
+            "whitespace=   ",
+            "   =5.4.56",
+            "description = hello there",
+            "=",
+            "",
+            " ",
+            " = ",
+            "etc=/hosts/be=gin",
+            "etc = /hosts/stuff",
+            "======",
+            "= = = =    ",
+            "    name = value = stuff = hello");
+
+    private static final Map.Entry EXPECTED_MAP_ENTRY_1 = new AbstractMap.SimpleEntry<>("name",
+            "factoryName");
+
+    private static final Map.Entry EXPECTED_MAP_ENTRY_2 = new AbstractMap.SimpleEntry<>(
+            "description",
+            "hello there");
+
+    private static final Map.Entry EXPECTED_MAP_ENTRY_3 = new AbstractMap.SimpleEntry<>("etc",
+            "/hosts/stuff");
+
+    private static final int EXPECTED_MAP_SIZE = 3;
+
+    private KeyValueParser parser;
+
+    @Before
+    public void setup() throws Exception {
+        parser = new KeyValueParser();
+    }
+
+    @After
+    public void destroy() throws Exception {
```

Comment:
Is there a reason you explicitly defined this despite it being an empty method? Checkstyle?